core/field_number.js

```diff
@@ -32,23 +32,40 @@ goog.require('goog.userAgent');
 
 
 /**
+ * Return an appropriate restrictor, depending on whether this FieldNumber
+ * allows decimal or negative numbers.
+ * @param {boolean} decimalAllowed Whether number may have decimal/float component.
+ * @param {boolean} negativeAllowed Whether number may be negative.
+ * @return {!RegExp} Regular expression for this FieldNumber's restrictor.
+ */
+var getNumRestrictor = function(decimalAllowed, negativeAllowed) {
+  var pattern = "[\\d]|[e]"; // Always allow digits and e, from Scratch 2.0.
+  if (decimalAllowed) {
+    pattern += "|[\\.]";
+  }
+  if (negativeAllowed) {
+    pattern += "|[-]";
+  }
+  return new RegExp(pattern);
+};
+
+/**
  * Class for an editable number field.
  * @param {string} text The initial content of the field.
+ * @param {boolean} decimalAllowed Whether number may have decimal/float component.
+ * @param {boolean} negativeAllowed Whether number may be negative.
  * @param {Function=} opt_validator An optional function that is called
  *     to validate any constraints on what the user entered.  Takes the new
  *     text as an argument and returns the accepted text or null to abort
  *     the change.
- * @param {number} precision Precision of the decimal value (negative power of 10).
- * @param {number} min Minimum value of the number.
- * @param {number} max Maximum value of the number.
  * @extends {Blockly.FieldTextInput}
  * @constructor
  */
-Blockly.FieldNumber = function(text, opt_validator, precision, min, max) {
-  this.precision_ = precision;
-  this.min_ = min;
-  this.max_ = max;
-  Blockly.FieldNumber.superClass_.constructor.call(this, text, opt_validator);
+Blockly.FieldNumber = function(text, decimalAllowed, negativeAllowed, opt_validator) {
```

Comment:
On the Android side we've implemented the [extra JSON parameters](https://developers.google.com/blockly/guides/create-custom-blocks/define-blocks#number_field) min, max, and precision to support these cases. This also allows for numbers that step by more than 1 at a time or have a specific range.

Would you mind using those instead of creating new ones?